win32.c

```diff
@@ -62,28 +63,40 @@ enet_time_set (enet_uint32 newTimeBase)
 int
 enet_address_set_host (ENetAddress * address, const char * name)
 {
-    struct hostent * hostEntry;
-
-    hostEntry = gethostbyname (name);
-    if (hostEntry == NULL ||
-        hostEntry -> h_addrtype != AF_INET)
-    {
-        unsigned long host = inet_addr (name);
-        if (host == INADDR_NONE)
-            return -1;
-        address -> host = host;
-        return 0;
-    }
-
-    address -> host = * (enet_uint32 *) hostEntry -> h_addr_list [0];
-
-    return 0;
+	struct sockaddr_in *sockaddr_ipv4 = NULL;
+	struct addrinfo *result = NULL;
+	struct addrinfo *ptr = NULL;
+	struct addrinfo infoAddr;
+	DWORD dwRetval = 0;
+
+	dwRetval = getaddrinfo(name, NULL, NULL, &result);
+	if (dwRetval != 0) {
+		ZeroMemory(&infoAddr, sizeof(infoAddr));
+		if (InetPton(AF_INET, name, &infoAddr) != 1) {
+			return -1;
+		}
+		ptr = &infoAddr;
+	} else {
+		ptr = result;
+	}
+	for (; ptr != NULL; ptr = ptr->ai_next) {
+		sockaddr_ipv4 = (struct sockaddr_in *) ptr->ai_addr;
+		break;
+	}
+	if (sockaddr_ipv4 == NULL) {
+		return -1;
+	}
+	address->host = (enet_uint32)sockaddr_ipv4->sin_addr.S_un.S_addr;
+
+	return 0;
 }
 
 int
 enet_address_get_host_ip (const ENetAddress * address, char * name, size_t nameLength)
 {
-    char * addr = inet_ntoa (* (struct in_addr *) & address -> host);
+	char buff[16];
+	const char * addr = InetNtop(AF_INET, (struct in_addr *)&address->host, buff, sizeof(buff));
```

Comment:
InetNtop is only available in Vista and later: https://msdn.microsoft.com/en-us/library/windows/desktop/cc805843(v=vs.85).aspx

Reply:
2024 calling. I don't think that Windows XP support and older is any kind of argument anymore, so this could probably be moved forward one away or another.